src/app/components/RuntimeEvents/EventListFilterSwitch.tsx

```diff
@@ -0,0 +1,71 @@
+import { FC } from 'react'
+import Chip from '@mui/material/Chip'
+import CheckIcon from '@mui/icons-material/Check'
+import { useTranslation } from 'react-i18next'
+import { COLORS } from '../../../styles/theme/colors'
+import { TFunction } from 'i18next'
+import Box from '@mui/material/Box'
+
+export const EventFilterMode = {
+  All: 'all',
+  NonTX: 'nonTX',
+} as const
+
+const eventFilterModes: EventFilterMode[] = [EventFilterMode.All, EventFilterMode.NonTX]
+
+const getEventFilterModeName = (t: TFunction, mode: EventFilterMode): string => {
+  switch (mode) {
+    case 'all':
+      return t('runtimeEvent.filter.all')
+    case 'nonTX':
+      return t('runtimeEvent.filter.nonTx')
+  }
+}
+
+const Pill: FC<{ label: string; selected: boolean; onSelect: () => void }> = ({
+  label,
+  selected,
+  onSelect,
+}) => {
+  return (
+    <Chip
+      icon={selected ? <CheckIcon htmlColor={'#FFFFFF'} /> : undefined}
+      label={label}
+      onClick={onSelect}
+      sx={{
+        display: 'flex',
+        height: 30,
+        padding: '3px 10px',
+        alignItems: 'center',
+        borderRadius: 9,
+        border: `1px solid ${COLORS.brandMedium}`,
+        background: selected ? COLORS.brandMedium : COLORS.brandLightBlue,
+        color: selected ? 'white' : 'black',
+      }}
+    />
+  )
+}
+
+// eslint-disable-next-line @typescript-eslint/no-redeclare
+export type EventFilterMode = (typeof EventFilterMode)[keyof typeof EventFilterMode]
```

Comment:
How does this + `export const EventFilterMode = {` work

Reply:
On is a type, the other is a const. TS compiler is smart enough to figure out which one we want in each situation. (The type is actually eliminated during the transpilation phase, so in the JS we only have the const.)

In any case, this is not something I came up with here, we already have a similar solution for `Layer` and other things, too.